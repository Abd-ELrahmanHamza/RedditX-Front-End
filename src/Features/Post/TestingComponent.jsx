// Import contexts providers
import { SubmitDestinationProvider } from "Features/Post/Contexts/submitDestination";
import { CreatePostTitleProvider } from "Features/Post/Contexts/createPostTitle";
import { CreatePostFlagsProvider } from "Features/Post/Contexts/createPostFlags";
import { CreatePostAttachmentsProvider } from "Features/Post/Contexts/createPostAttachments";
import { CreatePostFlairsProvider } from "Features/Post/Contexts/createPostFlairs";
import { CreatePostTextProvider } from "Features/Post/Contexts/createPostText";
import {
  AuthProvider,
  useAuth,
} from "Features/Authentication/Contexts/Authentication";

// Import react router dom
import { BrowserRouter, Route, Routes } from "react-router-dom";

// Import themes
import darkTheme from "Theme/darkTheme";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";

// Import theme provider from styled components
import { ThemeProvider } from "styled-components";
import { useEffect } from "react";

const AuthTesting = ({ children }) => {
  const auth = useAuth();
  useEffect(() => {
    auth.login({
      expiresIn: 86400,
      token:
        "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbFR5cGUiOiJiYXJlIGVtYWlsIiwidXNlcm5hbWUiOiJ0Ml9ob21lIiwiaWF0IjoxNjY5OTI0MDEzLCJleHAiOjE2NzAzNTYwMTN9.FDa59vJi5gWy4dQVkEtgyCfm3Pp2BtHkPz0_AfNVKgY",
      username: "t2_home",
    });
  }, []);

  return (
    <BrowserRouter>
      <ThemeProvider theme={{ ...defaultTheme, ...lightTheme }}>
        <SubmitDestinationProvider>
          <CreatePostTitleProvider>
            <CreatePostFlagsProvider>
              <CreatePostAttachmentsProvider>
                <CreatePostFlairsProvider>
                  <CreatePostTextProvider>{children}</CreatePostTextProvider>
                </CreatePostFlairsProvider>
              </CreatePostAttachmentsProvider>
            </CreatePostFlagsProvider>
          </CreatePostTitleProvider>
        </SubmitDestinationProvider>
      </ThemeProvider>
    </BrowserRouter>
  );
};

const TestingComponent = ({ children }) => {
  return (
    <AuthProvider>
      <AuthTesting>{children}</AuthTesting>
    </AuthProvider>
  );
};

export default TestingComponent;
