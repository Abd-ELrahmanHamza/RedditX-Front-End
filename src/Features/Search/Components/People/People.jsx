import PeopleCardItem from "../PeopleCardItem/PeopleCardItem";
import { PeopleContainer } from "./People.styled";

/**
 * Component that contains the People Card Elements.
 *
 * @Component
 * @returns {React.Component}
 */
const People = () => {
  return (
    <PeopleContainer>
      <p className="title">People</p>

      <PeopleCardItem />
      <PeopleCardItem />
      <PeopleCardItem />
      <PeopleCardItem />
    </PeopleContainer>
  );
};

export default People;
