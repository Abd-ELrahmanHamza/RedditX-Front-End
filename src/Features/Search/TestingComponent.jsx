// Import react router dom
import { BrowserRouter, Route, Routes } from "react-router-dom";
// Import themes
import darkTheme from "Theme/darkTheme";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";

// Import theme provider from styled components
import { ThemeProvider } from "styled-components";
import { SafeContextProvider } from "./Contexts/SafeSearchContext/Safe-context";
import { SearchContextProvider } from "./Contexts/SearchWordContext/Search-context";

const TestingComponent = ({ children }) => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={{ ...defaultTheme, ...lightTheme }}>
        <SearchContextProvider>
          <SafeContextProvider>{children}</SafeContextProvider>
        </SearchContextProvider>
      </ThemeProvider>
    </BrowserRouter>
  );
};

export default TestingComponent;
