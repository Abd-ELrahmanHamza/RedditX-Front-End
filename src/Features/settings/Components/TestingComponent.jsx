import {
  AuthProvider,
  useAuth,
} from "Features/Authentication/Contexts/Authentication";

// Import react router dom
import { BrowserRouter, Route, Routes } from "react-router-dom";

// Import themes
import darkTheme from "Theme/darkTheme";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";

// Import theme provider from styled components
import { ThemeProvider } from "styled-components";
import { useEffect } from "react";

const AuthTesting = ({ children }) => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={{ ...defaultTheme, ...lightTheme }}>
        {children}
      </ThemeProvider>
    </BrowserRouter>
  );
};

const TestingComponent = ({ children }) => {
  return (
    <AuthProvider>
      <AuthTesting>{children}</AuthTesting>
    </AuthProvider>
  );
};

export default TestingComponent;
