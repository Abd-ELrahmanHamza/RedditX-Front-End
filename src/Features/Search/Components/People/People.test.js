// import { shallow } from "enzyme";
import People from "./People";
// describe("People", () => {
//   it("this is a test for People", () => {
//     expect(shallow(<People />)).toMatchSnapshot();
//   });
// });
// import { shallow } from "enzyme";

// import People from "./People";
import { async } from "@firebase/util";
import { fireEvent, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import TestingComponent from "Features/Search/TestingComponent";

const PeopleList = [
  {
    avatar: "People_Image.jpg",
    userID: "t5_imagepro",
    about:
      "A subreddit dedicated to German photos and portraits from the period of 1933-1946 (dates are flexible)",
    username: "/u/Abdo",
    totalKarmas: 10,
  },
];
const PeopleFollow = [
  {
    id: "1",
  },
  {
    id: "t5_imagepro1",
  },
  {
    id: "t5_imagepro",
  },
  {
    id: "t5_imagepro2",
  },
];
describe("People component", () => {
  it('renders "People" in title', async () => {
    render(
      <TestingComponent>
        <People PeopleList={PeopleList} PeopleFollow={PeopleFollow} />
      </TestingComponent>
    );
    const outputElement = screen.getByText("People");
    expect(outputElement).toBeInTheDocument();
  });
});
