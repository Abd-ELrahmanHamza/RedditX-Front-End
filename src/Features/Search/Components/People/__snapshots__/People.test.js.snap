// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People Card Item this is a test for People Card Item 1`] = `ShallowWrapper {}`;

exports[`People this is a test for People 1`] = `ShallowWrapper {}`;
